File: ios/Memorezar/Data/Services/SupabaseConfig.swift
import Foundation

/// Configuration for Supabase REST API access
/// Admin: fill in projectURL and anonKey after creating your Supabase project
enum SupabaseConfig {
    // MARK: - Credentials (fill these in)

    static let projectURL = "https://tcaaozzijpgbaqzpnahl.supabase.co"
    static let anonKey = "sb_publishable_Cm0aEtK1Uv9-F7GIiU_15A_AgGH2ALL"

    // MARK: - Endpoints

    static var recordingsURL: URL {
        URL(string: "\(projectURL)/rest/v1/recordings")!
    }

    static var flagsURL: URL {
        URL(string: "\(projectURL)/rest/v1/flags")!
    }

    static var equivalencesURL: URL {
        URL(string: "\(projectURL)/rest/v1/equivalences")!
    }

    static var storageURL: URL {
        URL(string: "\(projectURL)/storage/v1/object/recordings")!
    }

    static func publicFileURL(path: String) -> URL {
        URL(string: "\(projectURL)/storage/v1/object/public/recordings/\(path)")!
    }

    // MARK: - Headers

    static var headers: [String: String] {
        let token = AuthService.shared.accessToken ?? anonKey
        return [
            "apikey": anonKey,
            "Authorization": "Bearer \(token)",
            "Content-Type": "application/json"
        ]
    }

    /// Whether credentials have been configured (not placeholder values)
    static var isConfigured: Bool {
        !projectURL.contains("YOUR_PROJECT_ID") && !anonKey.contains("YOUR_ANON_KEY")
    }
}

File: memorezar-android/app/src/main/java/com/memorezar/app/data/services/SupabaseConfig.kt
package com.memorezar.app.data.services

object SupabaseConfig {
    const val PROJECT_URL = "https://tcaaozzijpgbaqzpnahl.supabase.co"
    const val ANON_KEY = "sb_publishable_Cm0aEtK1Uv9-F7GIiU_15A_AgGH2ALL"

    // REST endpoints
    const val RECORDINGS_URL = "$PROJECT_URL/rest/v1/recordings"
    const val FLAGS_URL = "$PROJECT_URL/rest/v1/flags"
    const val EQUIVALENCES_URL = "$PROJECT_URL/rest/v1/equivalences"
    const val SUPPORT_TICKETS_URL = "$PROJECT_URL/rest/v1/support_tickets"
    const val PACKS_URL = "$PROJECT_URL/rest/v1/suggestion_packs"

    // Storage
    const val STORAGE_URL = "$PROJECT_URL/storage/v1/object/recordings"

    /** Private bucket for per-user audio backups. Owner-only RLS — files at
     *  <user_id>/<local_recording_id>.m4a. Used by CloudBackupService on
     *  Back Up Now / Restore. NOT for community recordings. */
    const val RECORDING_BACKUPS_URL = "$PROJECT_URL/storage/v1/object/recording-backups"

    fun publicFileURL(path: String): String =
        "$PROJECT_URL/storage/v1/object/public/recordings/$path"

    // Edge Functions
    const val TTS_URL = "$PROJECT_URL/functions/v1/tts"

    val isConfigured: Boolean
        get() = !PROJECT_URL.contains("YOUR_PROJECT_ID") && !ANON_KEY.contains("YOUR_ANON_KEY")
}
